modules/images/webp-uploads/generation.php

```diff
@@ -0,0 +1,250 @@
+<?php
+/**
+ * On demand generation functions.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Generates images on the fly for the registered image sizes and available transformations.
+ *
+ * @since n.e.x.t
+ */
+function webp_uploads_generate_missing_images() {
+	if ( ! is_404() ) {
+		return;
+	}
+
+	$uploads_dir = wp_upload_dir();
+	$baseurl     = wp_parse_url( $uploads_dir['baseurl'], PHP_URL_PATH );
+
+	// Bail if the current request is not to an uploaded image.
+	if ( substr( $_SERVER['REQUEST_URI'], 0, strlen( $baseurl ) ) !== $baseurl ) {
+		return;
+	}
```

Comment:
As far as I can tell, you're using the "regular" URLs that look like they point to within the `uploads` directory. That won't work for a core implementation, as several environments are configured to never route "real-looking" direct file URLs to PHP. We need to use some sort of dedicated endpoint for this, with its own URL, to make it reliable for core.